specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json

```diff
@@ -3714,6 +3726,58 @@
         }
       } 
     },
+    "SalesforceMarketingCloudLinkedService": { 
+      "x-ms-discriminator-value": "SalesforceMarketingCloud", 
+      "description": "Salesforce Marketing Cloud linked service.", 
+      "type": "object", 
+      "allOf": [ 
+        { 
+          "$ref": "#/definitions/LinkedService" 
+        } 
+      ], 
+      "properties": { 
+        "typeProperties": { 
+          "description": "Salesforce Marketing Cloud linked service properties.", 
+          "x-ms-client-flatten": true, 
+          "$ref": "#/definitions/SalesforceMarketingCloudLinkedServiceTypeProperties" 
+        } 
+      }, 
+      "required": [ 
+        "typeProperties" 
+      ] 
+    }, 
+    "SalesforceMarketingCloudLinkedServiceTypeProperties": { 
+      "description": "Salesforce Marketing Cloud linked service properties.", 
+      "properties": {
+        "clientId": {
+          "description": "The client ID associated with the Salesforce Marketing Cloud application.",
+          "type": "object"
```

Comment:
Please add the actual type of the property and do this for the other properties marked as object.

Reply:
Sure. Thanks for the comment.